### DISCUSSION TITLE
negative dimensions are not allowed error

### DISCUSSION THREAD

**Discussion by james6601** [Trusted-AI/adversarial-robustness-toolbox#2097] [Q&A]
Hi guys, I have trained an SVM , and when I try to evaluate the ART classifier on benign test samples, I receive the error  "negative dimensions are not allowed".

I thought maybe there would be a data mismatch between the features and labels I'm trying to make a prediction on, but when I run the following code, they are the same size

```
print(np.shape(test_features))
(5039, 38)

print(np.shape(test_labels))
(5039,) 
```

Just wondering if there is someone out there who can help me fix this error?

Thanks.

**Comment by beat-buesser** [comment]
Hi @james6601 Can you share the entire error message with stack trace?

**Reply by james6601** [reply]
Hi,

Thanks for getting back to me. The error along with the stack trace is shown below:


 ```
 File ~\OneDrive\Documents\ids\fgsm.py:50 in <module>
    predictions = classifier.predict(test_features)

  File ~\AppData\Roaming\Python\Python39\site-packages\art\estimators\classification\classifier.py:73 in replacement_function
    return fdict[func_name](self, *args, **kwargs)

  File ~\AppData\Roaming\Python\Python39\site-packages\art\estimators\classification\scikitlearn.py:1418 in predict
    one_hot_targets = np.eye(self.nb_classes)[targets]

  File C:\ProgramData\Anaconda3\lib\site-packages\numpy\lib\twodim_base.py:214 in eye
    m = zeros((N, M), dtype=dtype, order=order)

ValueError: negative dimensions are not allowed
```

**Reply by beat-buesser** [reply]
What is the shape of `test_features` and `train_labels`?

**Reply by james6601** [reply]
Hi again @beat-buesser, below are the shapes of test_features and train_labels:

```
print(np.shape(test_features))
(5039, 38)

print(np.shape(train_labels))
(20152,)
```

Also, this is what i get when i print the following:

```
In [4]: classifier.nb_classes 
Out[4]: -1
```
its as if the classifier doesnt recognise the labels im using so it gives the default of -1

**Reply by beat-buesser** [reply]
Can you try with `train_labels` being one-hot-encoded?
